c10/macros/Macros.h

```diff
@@ -109,6 +109,12 @@ namespace at { namespace cuda { using namespace c10::hip; }}
 #define C10_HOST_DEVICE __host__ __device__
 #define C10_DEVICE __device__
 #define C10_HOST __host__
+#define C10_MAX_THREADS_PER_BLOCK(val) (((val) <= 1024) ? (val) : (512))
+#if __CUDA_ARCH__ >= 750
+#define C10_MIN_BLOCKS_PER_SM(threads_per_block, blocks_per_sm) ((((threads_per_block)*(blocks_per_sm) <= 1024) ? (blocks_per_sm) : ((1024 + threads_per_block - 1) / threads_per_block)))
```

Comment:
Can we deduplicate these macros by simply defining a "max threads" constant, which varies depending on `__CUDA_ARCH__` and then using it in a single macro appropriately?

Reply:
Also, I want a comment saying what the macro actually computes. "C10_MIN_BLOCKS_PER_SM computes the minimum blocks per SM given a requested threads per block, and blocks per sm?" That's not quite fully formed, in my head.

Reply:
You're missing parentheses around `threads_per_block` at the end